cassandra-analytics-core/src/main/java/org/apache/cassandra/spark/bulkwriter/BulkSparkConf.java

```diff
@@ -240,38 +241,47 @@ protected Set<String> buildBlockedInstances(Map<String, String> options)
     {
-        int legacyOptionValue = -1;
-        if (options.containsKey(WriterOptions.SSTABLE_DATA_SIZE_IN_MB.name()))
-        {
-            LOGGER.warn("The writer option: SSTABLE_DATA_SIZE_IN_MB is deprecated. " +
-                        "Please use SSTABLE_DATA_SIZE_IN_MIB instead. See option description for details.");
-            legacyOptionValue = MapUtils.getInt(options, WriterOptions.SSTABLE_DATA_SIZE_IN_MB.name(),
-                                                DEFAULT_SSTABLE_DATA_SIZE_IN_MIB, "sstable data size in mebibytes");
-        }
+        return MapUtils.resolveDeprecated(options, WriterOptions.SSTABLE_DATA_SIZE_IN_MIB.name(), WriterOptions.SSTABLE_DATA_SIZE_IN_MB.name(), option -> {
+            if (option == null)
+            {
+                return DEFAULT_SSTABLE_DATA_SIZE_IN_MIB;
+            }
 
-        if (options.containsKey(WriterOptions.SSTABLE_DATA_SIZE_IN_MIB.name()))
-        {
-            LOGGER.info("The writer option: SSTABLE_DATA_SIZE_IN_MIB is defined. " +
-                        "Favor the value over SSTABLE_DATA_SIZE_IN_MB");
-            return MapUtils.getInt(options, WriterOptions.SSTABLE_DATA_SIZE_IN_MIB.name(),
-                                   DEFAULT_SSTABLE_DATA_SIZE_IN_MIB, "sstable data size in mebibytes");
+            return MapUtils.getInt(options, option, DEFAULT_SSTABLE_DATA_SIZE_IN_MIB, "sstable data size in mebibytes");
+        });
+    }
 
-        }
+    public static String resolveSidecarContactPoints(Map<String, String> options)
+    {
+        return MapUtils.resolveDeprecated(options, WriterOptions.SIDECAR_CONTACT_POINTS.name(), WriterOptions.SIDECAR_INSTANCES.name(), option -> {
+            if (option == null)
+            {
+                return null;
+            }
 
-        return legacyOptionValue == -1 ? DEFAULT_SSTABLE_DATA_SIZE_IN_MIB : legacyOptionValue;
+            return MapUtils.getOrDefault(options, option, null);
+        });
     }
 
     protected Set<? extends SidecarInstance> buildSidecarInstances()
     {
-        String[] split = Objects.requireNonNull(sidecarInstancesValue, "Unable to build sidecar instances from null value")
+        String[] split = Objects.requireNonNull(sidecarContactPointsCsv, "Unable to build sidecar instances from null value")
                                 .split(",");
         return Arrays.stream(split)
-                     .map(hostname -> new SidecarInstanceImpl(hostname, effectiveSidecarPort))
+                     .filter(StringUtils::isNotEmpty)
+                     .map(hostname -> SidecarInstanceFactory.createFromString(hostname, effectiveSidecarPort))
                      .collect(Collectors.toSet());
     }
 
+    // extract port from input if present and use the extract port to create SidecarInstanceImpl
+    // otherwise, the input does not contain port info, and it uses the default port
+    static SidecarInstanceImpl createSidecarInstance(String input, int defaultPort)
```

Comment:
unused method?

Reply:
forgot to merge after switch to `SidecarInstanceFactory`